From d40f5215deff3923632afe64d90b050453f2027a Mon Sep 17 00:00:00 2001
From: Hao Zheng <haoz@marvell.com>
Date: Fri, 21 Jun 2019 14:10:33 -0700
Subject: [PATCH 0213/1921] octeontx2-af: fix parser error for udp packets with
 no payload

fix npc parser error when parsing udp packets that don't have
any payload

Change-Id: Ifce2ca41fca67c3563c147b7cb6d19699e9fb888
Signed-off-by: Hao Zheng <haoz@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/11455
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
index bad1bde74482..5bdb83c4646b 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
@@ -8925,7 +8925,7 @@ static struct npc_kpu_profile_action kpu5_action_entries[] = {
 	},
 	{
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
-		2, 8, 10, 2, 0,
+		2, 0, 0, 2, 0,
 		NPC_S_KPU8_UDP, 20, 1,
 		NPC_LID_LC, NPC_LT_LC_IP,
 		0,
@@ -9181,7 +9181,7 @@ static struct npc_kpu_profile_action kpu5_action_entries[] = {
 	},
 	{
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
-		2, 8, 10, 2, 0,
+		2, 0, 0, 2, 0,
 		NPC_S_KPU8_UDP, 40, 1,
 		NPC_LID_LC, NPC_LT_LC_IP6,
 		0,
-- 
2.31.1

